src/uct/ib/base/ib_iface.c

```diff
@@ -177,7 +177,22 @@ int uct_ib_iface_is_reachable(const uct_iface_h tl_iface, const uct_device_addr_
     uint16_t lid;
 
     uct_ib_address_unpack(ib_addr, &lid, &is_global, &gid);
-    return (gid.global.subnet_prefix == iface->gid.global.subnet_prefix);
+
+    if ((ib_addr->flags & UCT_IB_ADDRESS_FLAG_GID_RAW) == 0) {
+        /* IB */
+        return (gid.global.subnet_prefix == iface->gid.global.subnet_prefix);
+    } else {
+        /* RoCE */
+        /* check that there is no lid and that the raw_gid is on */
+        if ((ib_addr->flags & UCT_IB_ADDRESS_FLAG_GID_RAW) &&
```

Comment:
why check ib_addr->flags & UCT_IB_ADDRESS_FLAG_GID_RAW again, if it was already checked in the previous 'if'?